extractors/bilibili/bilibili.go

```diff
@@ -55,44 +48,72 @@ func genAPI(aid, cid int, bangumi bool, quality, seasonType, cookie string) (str
 		}
 		utoken = t.Data.Token
 	}
+	var api string
 	if bangumi {
 		// The parameters need to be sorted by name
 		// qn=0 flag makes the CDN address different every time
-		// quality=116(1080P 60) is the highest quality so far
+		// quality=120(4k) is the highest quality so far
 		params = fmt.Sprintf(
-			"appkey=%s&cid=%d&module=bangumi&otype=json&qn=%s&quality=%s&season_type=%s&type=",
-			appKey, cid, quality, quality, seasonType,
+			"avid=%d&cid=%d&bvid=&qn=%d&type=&otype=json&fourk=1&fnver=0&fnval=16",
```

Comment:
Still need the avid?

Reply:
It seems the avid is no longer necessary.

Reply:
Can we have a test? If this is unnecessary, remove it and fill up the bvid field

Reply:
Both the bvid and avid field is unnecessary.
Proof:
```
go run main.go -i -d https://www.bilibili.com/bangumi/play/ep94730

URL:         https://www.bilibili.com/bangumi/play/ep94730
Method:      GET
Headers:     http.Header{
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Referer":         {"https://www.bilibili.com"},
}
Status Code: 200

URL:         https://api.bilibili.com/pgc/player/web/playurl?cid=9191764&bvid=&qn=120&type=&otype=json&fourk=1&fnver=0&fnval=16
Method:      GET
Headers:     http.Header{
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Referer":         {"https://www.bilibili.com"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
}
Status Code: 200

URL:         https://api.bilibili.com/pgc/player/web/playurl?cid=9191764&bvid=&qn=64&type=&otype=json&fourk=1&fnver=0&fnval=16
Method:      GET
Headers:     http.Header{
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Referer":         {"https://www.bilibili.com"},
}
Status Code: 200
......
......
 Site:      哔哩哔哩 bilibili.com
 Title:     一人之下：第5话 祸福无门，惟人自召
 Type:      video
 Streams:   # All available quality
     [64]  -------------------
     Quality:         高清 720P
     Size:            237.87 MiB (249424400 Bytes)
     # download with: annie -f 64 ...

     [32]  -------------------
     Quality:         清晰 480P
     Size:            151.08 MiB (158421336 Bytes)
     # download with: annie -f 32 ...

     [16]  -------------------
     Quality:         流畅 360P
     Size:            79.05 MiB (82893645 Bytes)
     # download with: annie -f 16 ...
```

```
go run main.go -i -d https://www.bilibili.com/bangumi/play/ss12044
URL:         https://www.bilibili.com/bangumi/play/ss12044
Method:      GET
Headers:     http.Header{
    "Referer":         {"https://www.bilibili.com"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
}
Status Code: 200

URL:         https://api.bilibili.com/pgc/player/web/playurl?cid=126654047&bvid=&qn=120&type=&otype=json&fourk=1&fnver=0&fnval=16
Method:      GET
Headers:     http.Header{
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Referer":         {"https://www.bilibili.com"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
}
Status Code: 200

URL:         https://api.bilibili.com/pgc/player/web/playurl?cid=126654047&bvid=&qn=112&type=&otype=json&fourk=1&fnver=0&fnval=16
Method:      GET
Headers:     http.Header{
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Referer":         {"https://www.bilibili.com"},
}
Status Code: 200

URL:         https://api.bilibili.com/pgc/player/web/playurl?cid=126654047&bvid=&qn=80&type=&otype=json&fourk=1&fnver=0&fnval=16
Method:      GET
Headers:     http.Header{
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Referer":         {"https://www.bilibili.com"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
}
Status Code: 200
......
......
 Site:      哔哩哔哩 bilibili.com
 Title:     你的名字。
 Type:      video
 Streams:   # All available quality
     [80]  -------------------
     Quality:         高清 1080P
     Size:            1439.41 MiB (1509326767 Bytes)
     # download with: annie -f 80 ...

     [64]  -------------------
     Quality:         高清 720P
     Size:            1105.35 MiB (1159045616 Bytes)
     # download with: annie -f 64 ...

     [32]  -------------------
     Quality:         清晰 480P
     Size:            703.74 MiB (737921007 Bytes)
     # download with: annie -f 32 ...

     [16]  -------------------
     Quality:         流畅 360P
     Size:            308.90 MiB (323902449 Bytes)
     # download with: annie -f 16 ...
```